synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -0,0 +1,229 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file dialogs/canvasresize.cpp
+**	\brief Template File
+**
+**	$Id$
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include <ETL/handle>
+#include <dialogs/canvasproperties.h>
+#include <gui/app.h>
+#include <gui/localization.h>
+#include <gui/resourcehelper.h>
+#include <widgets/widget_link.h>
+#include <synfig/renddesc.h>
+#include <synfigapp/canvasinterface.h>
+
+#include "canvasresize.h"
```

Comment:
Please, place the #include of the header of the .cpp first (before #include <ETL/handle>)

Reply:
Ok, on it.

Reply:
Build failed. I think cpp header file was supposed to be after ETL header because I used forward declaration over here:
![Screenshot from 2021-01-02 12-04-03](https://user-images.githubusercontent.com/24214509/103452193-a80aaa80-4cf2-11eb-879c-5b2a95a17bc3.png)

Reply:
Reverting...

Reply:
Build succeeded :)

Reply:
> Build failed. I think cpp header file was supposed to be after ETL header because I used forward declaration over here:
> ![Screenshot from 2021-01-02 12-04-03](https://user-images.githubusercontent.com/24214509/103452193-a80aaa80-4cf2-11eb-879c-5b2a95a17bc3.png)

As it is #include'd in synfig-studio/src/gui/canvasview.h (and only there beside implementation file) it shouldn't be a problem to #include <ETL/handle> directly in header file.

Reply:
> it shouldn't be a problem to #include <ETL/handle> directly in header file.

That would mean removing forward class declaration in the above header file. I guess consistency with placing our declaration file first is more suiting.

I'll make the changes here from my side :)

Reply:
Sorry for the long-delayed reply. Thought all reviews were resolved.